### DISCUSSION TITLE
Error using GPU with the accuracy-pipeline (AttributeError: module 'torch._C' has no attribute '_cuda_setDevice')

### DISCUSSION THREAD

**Discussion by mscherrmann** [explosion/spaCy#7358] [Help: Installation] [answered]
## How to reproduce the behaviour

I created a virtual environment and installed spacy with the following steps:

1. conda create --name spacy python=3.6
2. conda activate spacy
3. pip install -U pip setuptools wheel
4. pip install -U spacy[cuda102]
5. python -m spacy download en_core_web_trf

Then I tried to run the following code in the console within the created virtual environment:
`import spacy`
`spacy.require_gpu()`

I get the following error message:

> Traceback (most recent call last):
>
> File "<ipython-input-2-ffa7a905abff>", line 1, in <module>
>   spacy.require_gpu()
>
>  File "C:\ProgramData\Anaconda3\envs\spacy\lib\site-packages\thinc\util.py", line 184, in require_gpu
>   set_active_gpu(gpu_id)
>
>  File "C:\ProgramData\Anaconda3\envs\spacy\lib\site-packages\thinc\util.py", line 151, in set_active_gpu
>    torch.cuda.set_device(gpu_id)
>
>  File "C:\ProgramData\Anaconda3\envs\spacy\lib\site-packages\torch\cuda\__init__.py", line 263, in set_device
>    torch._C._cuda_setDevice(device)
>
>AttributeError: module 'torch._C' has no attribute '_cuda_setDevice'

Note:

- The code runs fine when I execute it **before** running step 5. in the installation process
- Spacy works fine if I run it on CPU instead of GPU
- If I install the efficiency-pipeline instead, both the CPU and GPU versions work fine

I expect that there is some update of torch when installing the accuracy pipeline using "python -m spacy download en_core_web_trf". The console outputs among others:

Collecting torchcontrib<0.1.0,>=0.0.2
  Using cached torchcontrib-0.0.2-py3-none-any.whl
Collecting ftfy<6.0.0,>=5.0.0
  Using cached ftfy-5.9-py3-none-any.whl
Collecting spacy-alignments<1.0.0,>=0.7.2
  Using cached spacy_alignments-0.7.2-cp36-cp36m-win_amd64.whl (183 kB)
Collecting transformers<4.3.0,>=3.1.0
  Using cached transformers-4.2.2-py3-none-any.whl (1.8 MB)
Collecting torch>=1.5.0
  Using cached torch-1.7.1-cp36-cp36m-win_amd64.whl (184.1 MB)

In the attachment you will find the installed packages of the virtual environment:
[venv_packages.txt](https://github.com/explosion/spaCy/files/6008977/venv_packages.txt)

What can I do to fix this error? Thank you a lot for your help in advance.

## Your Environment

* Operating System: Windows 10
* Python Version Used: 3.6.12
* spaCy Version Used: 3.0.3
* Pipelines: en-core-web-trf 3.0.0 
* Environment Information: Cuda 10.2 (output of `nvcc -V`), Anaconda 2020.07, Nvidia GeForce 2080Ti

**Comment by adrianeboyd** [comment, marked as answer]
Hi, it sounds like something when wrong with the `torch` installation, which is unfortunately common with a plain `pip install torch`. (`torch` is being installed as a dependency of `en_core_web_trf`.)

Uninstall all the torch-related packages (or just create a new conda env) and try the directions here for using the pytorch quickstart, using conda instead of pip to install torch after step 2 above: https://spacy.io/usage/embeddings-transformers#transformers-installation

**Comment by mscherrmann** [comment]
Hi, thank you very much for your quick response. You are absolutly right, now everything works very well.

For completeness: The only change to the installation process described above is that I added the command 
`conda install pytorch torchvision torchaudio cudatoolkit=10.2 -c pytorch` between steps 2. and 3.
